src/library/scala/collection/immutable/Vector.scala

```diff
@@ -799,98 +727,102 @@ private[immutable] trait VectorPointer[T] {
-      } else
-      if (xor < (1 << 15)) { // level = 2
-        display2((index >> 10) & 31).asInstanceOf[Array[AnyRef]]((index >> 5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
-      } else
-      if (xor < (1 << 20)) { // level = 3
-        display3((index >> 15) & 31).asInstanceOf[Array[AnyRef]]((index >> 10) & 31).asInstanceOf[Array[AnyRef]]((index >> 5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
-      } else
-      if (xor < (1 << 25)) { // level = 4
-        display4((index >> 20) & 31).asInstanceOf[Array[AnyRef]]((index >> 15) & 31).asInstanceOf[Array[AnyRef]]((index >> 10) & 31).asInstanceOf[Array[AnyRef]]((index >> 5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
-      } else
-      if (xor < (1 << 30)) { // level = 5
-        display5((index >> 25) & 31).asInstanceOf[Array[AnyRef]]((index >> 20) & 31).asInstanceOf[Array[AnyRef]]((index >> 15) & 31).asInstanceOf[Array[AnyRef]]((index >> 10) & 31).asInstanceOf[Array[AnyRef]]((index >> 5) & 31).asInstanceOf[Array[AnyRef]](index & 31).asInstanceOf[T]
-      } else { // level = 6
+      if        (xor < (1 <<  5)) { // level = 0
+        val array = display0
+        array(index & 31).asInstanceOf[T]
+      } else if (xor < (1 << 10)) { // level = 1
+        var array = display1
+        array = array((index >>>  5) & 31).asInstanceOf[Array[AnyRef]]
+        array(index & 31).asInstanceOf[T]
+      } else if (xor < (1 << 15)) { // level = 2
+        var array = display2
+        array = array((index >>> 10) & 31).asInstanceOf[Array[AnyRef]]
+        array = array((index >>>  5) & 31).asInstanceOf[Array[AnyRef]]
+        array(index & 31).asInstanceOf[T]
+      } else if (xor < (1 << 20)) { // level = 3
+        var array = display3
+        array = array((index >>> 15) & 31).asInstanceOf[Array[AnyRef]]
+        array = array((index >>> 10) & 31).asInstanceOf[Array[AnyRef]]
+        array = array((index >>>  5) & 31).asInstanceOf[Array[AnyRef]]
+        array(index & 31).asInstanceOf[T]
+      } else if (xor < (1 << 25)) { // level = 4
+        var array = display4
+        array = array((index >>> 20) & 31).asInstanceOf[Array[AnyRef]]
+        array = array((index >>> 15) & 31).asInstanceOf[Array[AnyRef]]
+        array = array((index >>> 10) & 31).asInstanceOf[Array[AnyRef]]
+        array = array((index >>>  5) & 31).asInstanceOf[Array[AnyRef]]
+        array(index & 31).asInstanceOf[T]
+      } else if (xor < (1 << 30)) { // level = 5
+        var array = display5
+        array = array((index >>> 25) & 31).asInstanceOf[Array[AnyRef]]
+        array = array((index >>> 20) & 31).asInstanceOf[Array[AnyRef]]
+        array = array((index >>> 15) & 31).asInstanceOf[Array[AnyRef]]
+        array = array((index >>> 10) & 31).asInstanceOf[Array[AnyRef]]
+        array = array((index >>>  5) & 31).asInstanceOf[Array[AnyRef]]
+        array(index & 31).asInstanceOf[T]
+      } else {                      // level = 6
```

Comment:
Done, resolved it via parenthesizing everything to make it multilinable, making alignment trivial also :)